types/lib/mixins/properties-changed.d.ts

```diff
@@ -0,0 +1,257 @@
+   * deserialized via `attributeChangedCallback` and set to the associated
+   * property using `dash-case`-to-`camelCase` convention.
+   */
+  function PropertiesChanged<T extends new(...args: any[]) => {}>(base: T): {
+    new(...args: any[]): PropertiesChanged
+  } & T
+
+  interface PropertiesChanged {
+
+    /**
+     * Creates a setter/getter pair for the named property with its own
+     * local storage.  The getter returns the value in the local storage,
+     * and the setter calls `_setProperty`, which updates the local storage
+     * for the property and enqueues a `_propertiesChanged` callback.
+     *
+     * This method may be called on a prototype or an instance.  Calling
+     * this method may overwrite a property value that already exists on
+     * the prototype/instance by creating the accessor.
+     *
+     * @param property Name of the property
+     * @param readOnly When true, no setter is created; the
+     *   protected `_setProperty` function must be used to set the property
+     */
+    _createPropertyAccessor(property: string, readOnly?: boolean): any;
+
+    /**
+     * Defines a property accessor for the given property.
+     *
+     * @param property Name of the property
+     * @param readOnly When true, no setter is created
+     */
+    _definePropertyAccessor(property: string, readOnly?: boolean): any;
+
+    /**
+     * Lifecycle callback called when properties are enabled via
+     * `_enableProperties`.
+     *
+     * Users may override this function to implement behavior that is
+     * dependent on the element having its property data initialized, e.g.
+     * from defaults (initialized from `constructor`, `_initializeProperties`),
+     * `attributeChangedCallback`, or values propagated from host e.g. via
+     * bindings.  `super.ready()` must be called to ensure the data system
+     * becomes enabled.
+     */
+    ready(): any;
+
+    /**
+     * Initializes the local storage for property accessors.
+     *
+     * Provided as an override point for performing any setup work prior
+     * to initializing the property accessor system.
+     */
+    _initializeProperties(): any;
+
+    /**
+     * Called at ready time with bag of instance properties that overwrote
+     * accessors when the element upgraded.
+     *
+     * The default implementation sets these properties back into the
+     * setter at ready time.  This method is provided as an override
+     * point for customizing or providing more efficient initialization.
+     *
+     * @param props Bag of property values that were overwritten
+     *   when creating property accessors.
+     */
+    _initializeInstanceProperties(props: object|null): any;
+
+    /**
+     * Updates the local storage for a property (via `_setPendingProperty`)
+     * and enqueues a `_proeprtiesChanged` callback.
+     *
+     * @param property Name of the property
+     * @param value Value to set
+     */
+    _setProperty(property: string, value: any): any;
```

Comment:
should return `void`

Reply:
Fixed by Tim in https://github.com/Polymer/polymer/pull/4998